api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceDaoTest.java

```diff
@@ -328,6 +332,128 @@ public void findNamespacesByActiveStatusAndFirecloudUuidsIn_multiple() {
+            List.of("orphaned-namespace-1", "orphaned-namespace-2") // expected orphaned
+            ),
+        Arguments.of(
+            "Single orphaned workspace",
+            List.of("orphaned-namespace"), // namespaces in db
+            List.of("externally-referenced-namespace"), // externally referenced namespaces
+            List.of("orphaned-namespace") // expected orphaned
+            ),
+        Arguments.of(
+            "Multiple orphaned workspaces",
+            List.of("orphaned-namespace-1", "orphaned-namespace-2"), // namespaces in db
+            List.of(
+                "externally-referenced-namespace-1",
+                "externally-referenced-namespace-2"), // externally referenced namespaces
+            List.of("orphaned-namespace-1", "orphaned-namespace-2") // expected orphaned
+            ),
+        Arguments.of(
+            "No orphaned workspaces",
+            List.of("namespace-1", "namespace-2"), // namespaces in db
+            List.of("namespace-1", "namespace-2"), // externally referenced namespaces
+            Collections.emptyList() // expected orphaned
+            ));
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("orphanedWorkspaceScenarios")
+  void testFindAllOrphanedWorkspaceNamespaces_basicScenarios(
+      String scenarioName,
+      List<String> workspaceNamespaces,
+      List<String> externallyReferencedNamespaces,
+      List<String> expectedOrphaned) {
+    workspaceDao.deleteAll();
+
+    // Create and save all workspaces
+    for (String namespace : workspaceNamespaces) {
+      createAndSaveWorkspace(namespace, WorkspaceActiveStatus.ACTIVE);
+    }
+
+    // Execute the query
+    List<String> orphanedNamespaces =
+        workspaceDao.findAllOrphanedWorkspaceNamespaces(externallyReferencedNamespaces);
+
+    // Assert the results
+    if (expectedOrphaned.isEmpty()) {
+      assertThat(orphanedNamespaces).isEmpty();
+    } else {
+      assertThat(orphanedNamespaces).containsExactlyElementsIn(expectedOrphaned);
+    }
+  }
+
+  @Test
+  public void findAllOrphanedWorkspaceNamespaces_excludesDeletedWorkspaces() {
+    workspaceDao.deleteAll();
+
+    String deletedNamespace = "deleted-namespace";
+    String orphanedNamespace = "orphaned-namespace";
+    String externallyReferencedNamespace = "known-namespace";
+
+    createAndSaveWorkspace(deletedNamespace, WorkspaceActiveStatus.DELETED);
+    createAndSaveWorkspace(orphanedNamespace, WorkspaceActiveStatus.ACTIVE);
+    createAndSaveWorkspace(externallyReferencedNamespace, WorkspaceActiveStatus.ACTIVE);
+
+    List<String> orphanedNamespaces =
+        workspaceDao.findAllOrphanedWorkspaceNamespaces(List.of(externallyReferencedNamespace));
+
+    // Should only return the active orphaned workspace, not the deleted one
+    assertThat(orphanedNamespaces).containsExactly(orphanedNamespace);
+  }
+
+  @Test
+  public void findAllOrphanedWorkspaceNamespaces_duplicateNamespaces() {
```

Comment:
I am not sure if this test is necessary, but based on our current db configuration, it is possible to have two workspaces with the same namespaces.